src/components/NumberInput/NumberInput.stories.tsx

```diff
@@ -0,0 +1,47 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { Meta, StoryFn } from '@storybook/react';
+import { NumberInput } from './NumberInput';
+import { NumberInputProps } from './types';
+import React, { useState } from 'react';
+import { IconNook16 } from '@foundation/Icon';
+
+export default {
+    title: 'Components/Number Input',
+    component: NumberInput,
+    tags: ['autodocs'],
+    args: {},
+    argTypes: {
+        error: { type: 'boolean' },
```

Comment:
I don't think that having props like `error: boolean` and `errorText: string` would work as we have different type of validation for that Input with different colors and icons. We would need to add Success for sure and probably warning as a minimum.

Reply:
This goes back to the conversation of where do these props live.  If we move them to the `FormField` then I will just remove these completely and we can consider this question for that component.

Reply:
Agree with removing `error` and putting it in FormField
```tsx
<FormField
    name="myFieldName"
    render={() => (
      <FormItem>
        <FormLabel>Age</FormLabel>

        <FormControl>
          <NumberInput placeholder="123" {...field} />
        </FormControl>

        <FormDescription>This is a description to the form field</FormDescription>

        <FormMessage /> {/* Smart component which display the form field status (`error`, `success`, `pending`, ...) */}
      </FormItem>
    )}
  />
```

Reply:
We still the Input to know which validation status we're in. As the border color and the proper icon needs to be updated within it. 
![Screenshot 2023-09-04 at 10 53 28](https://github.com/Frontify/fondue/assets/35405438/06b44239-4521-4712-b718-23f69722b862)

![Screenshot 2023-09-04 at 10 54 44](https://github.com/Frontify/fondue/assets/35405438/0b0fdd23-8b9c-4c26-b6ff-cd3e3094cf6f)

Reply:
Agree with @imoutaharik At a minimum there will need to be some prop that tells the input what status icon/color to render.  Also should we enable the ability to pass a preferred icon for specific status'?

Reply:
No, in this case, we enforce the icon. Otherwise we will lose consistency for our inputs validation.

Reply:
The formfield should just pass the status to the field to add the border/icon thus the `{...field}` on the NumberInput ;)

Reply:
For now I will add a `status: Validation` prop and we can revisit once we have the correct structure of the new `FormField` component.